Sources/Datadog/Core/Feature.swift

```diff
@@ -120,6 +120,19 @@ internal struct FeatureStorage {
         self.reader = reader
         self.arbitraryAuthorizedWriter = arbitraryAuthorizedWriter
     }
+
+#if DD_SDK_COMPILED_FOR_TESTING
+    /// Flushes all async write operations.
+    ///
+    /// This method is executed synchronously. After return, the storage feature has no more
+    /// pending asynchronous write operations so all its data is ready for upload.
+    func flush() {
+        let consentAwareDataWriter = writer as? ConsentAwareDataWriter
+        let arbitraryDataWriter = arbitraryAuthorizedWriter as? ArbitraryDataWriter
+        let readWriteQueue = consentAwareDataWriter?.readWriteQueue ?? arbitraryDataWriter?.readWriteQueue
+        readWriteQueue?.sync {}
```

Comment:
I don't see where you are stopping this, I guess you are actually stopping the feature. This means the feature needs to be stopped first to avoid having a new event in the pipeline that could happen to bot be stored by the time your process exits.

Reply:
You're right 👍. I made the assumption that no data will be written (through logger, tracer or RUM monitor) after deinitializing the SDK. It doesn't need to be true, so I took more time on this PR and updated it in https://github.com/DataDog/dd-sdk-ios/pull/498/commits/9e0d0e4fdf3ad67069da04f916e8b780caad64a7 (👀 🙏).

Now the "storage" is deinitialized by:
```swift
writer.flushAndCancelSynchronously()
arbitraryAuthorizedWriter.flushAndCancelSynchronously()
reader.markAllFilesAsReadable()
```
which:
- sets the `isCancelled` flag in both async writers, making them act as no-op,
- sets the `ignoreFilesAgeWhenReading` flag in files orchestrator to skip file age condition for reading.

This way on `.deinitialize()` no more data can be saved, but everything can be uploaded. The consistency is additionally checked with `asserts()` for strict control in tests and no flakiness. Everything is conditionally compiled only for tests - no logic change in production code.